lib/semantics/check-do-concurrent.cc

```diff
@@ -299,10 +300,21 @@ static CS GatherLocalVariableNames(
   }
   return names;
 }
+
 static CS GatherReferencesFromExpression(const parser::Expr &expression) {
-  GatherSymbols gatherSymbols;
-  parser::Walk(expression, gatherSymbols);
-  return gatherSymbols.symbols;
+  // Use the new expression traversal framework if possible, for testing.
+  if (expression.typedExpr.has_value()) {
```

Comment:
It looks we would want expressions to be fully resolved before doing this. So perhaps expression resolution should be in a separate pass than these checks in PR #315.

Reply:
Sure.  Once that's assured, this code can be simplified.